sql/core/src/main/scala/org/apache/spark/sql/artifact/ArtifactManager.scala

```diff
@@ -279,6 +283,16 @@ class ArtifactManager(session: SparkSession) extends Logging {
     loader
   }
 
+  private[sql] def clone(newSession: SparkSession): ArtifactManager = {
+    val newArtifactManager = new ArtifactManager(newSession)
+    if (artifactPath.toFile.exists()) {
+      FileUtils.copyDirectory(artifactPath.toFile, newArtifactManager.artifactPath.toFile)
+    }
+    newArtifactManager.jarsList.addAll(jarsList)
+    newArtifactManager.pythonIncludeList.addAll(pythonIncludeList)
```

Comment:
Wouldn't the `pythonIncludeList` also require the `target -> normalizedRemoteRelativePath` change similar to `jarsList`?

Reply:
For Python we're adding only the filename to the list:
```
pythonIncludeList.add(target.getFileName.toString)
```

Reply:
Ah right!